.github/workflows/release.yaml

```diff
@@ -0,0 +1,76 @@
+name: Release
+
+on:
+    workflow_dispatch:
+        inputs:
+            new_version:
+              type: string
+              required: true
+            image_tag:
+              type: string
+              required: true
+            python_version:
+              type: string
+              default: "3.10"
+              required: true
+            poetry_version:
+              type: string
+              default: "1.5.1"
+              required: true
+
+jobs:
+    release:
+        runs-on: ubuntu-latest
+        permissions:
+            contents: write
+        steps:
+            - name: Checkout the repository
+              uses: actions/checkout@v3
+            - name: Change version in pyproject.toml
+              run: sed -i "s/\(^version = \"\)\(.*\)\"/\1${{ github.event.inputs.new_version }}\"/" pyproject.toml
+            - name: Install Python
+              uses: actions/setup-python@v4
+              with:
+                python-version: ${{ github.event.inputs.python_version }}
+            - name: Install Poetry
+              uses: abatilo/actions-poetry@v2
+              with:
+                poetry-version: ${{ github.event.inputs.poetry_version }}
+            - name: Run poetry install
+              run: poetry install --with docs
+            - name: Run poetry build
+              run: poetry build
+            - name: Create new documentation
+              run: poetry run pdoc --html -o docs src/codeflare_sdk && pushd docs && rm -rf cluster job utils && mv codeflare_sdk/* . && rm -rf codeflare_sdk && popd
+            - name: Set Pypi token
+              run: poetry config pypi-token.pypi ${{ secrets.PYPI_TOKEN }}
+            - name: Publish new release to Pypi repository
+              run: poetry publish
+
+            - name: Commit the pre-release changes
+              uses: stefanzweifel/git-auto-commit-action@v4
+              with:
+                file_pattern: 'docs pyproject.toml'
+                commit_message: "New release: ${{ github.event.inputs.new_version }}"
+            - name: Create Github release
+              uses: ncipollo/release-action@v1
+              with:
+                  tag: "v${{ github.event.inputs.new_version }}"
+
+            - name: Update SDK version in the Docker image
+              run: sed -i "s/\(.*codeflare-sdk==\)[^\\s]*\(.*\)/\1${{ github.event.inputs.new_version }}\2/" custom-nb-image/Dockerfile
+            - name: Image Build
+              run: |
+                cd custom-nb-image
+                docker build -t quay.io/project-codeflare/notebook:${{ github.event.inputs.image_tag }} .
+                docker tag quay.io/project-codeflare/notebook:${{ github.event.inputs.image_tag }} quay.io/project-codeflare/notebook:latest
```

Comment:
Can you separate creation of latest tag to be optional, like in https://github.com/project-codeflare/instascale/blob/e3df822968e6617cf5e2bcc60469268a42f55738/.github/workflows/instascale-release.yml#L62-L66 ?

The reason is that some releases may not be latest one (i.e. some patch release).

Reply:
Done. :+1: